tests/test_space.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""Test cases for functionality inside audiblelight/space.py"""
+
+import os
+
+import pytest
+import numpy as np
+from trimesh import Trimesh
+
+from audiblelight.space import Space, load_mesh, validate_mesh
+from audiblelight import utils
+
+
+TEST_RESOURCES = os.path.join(utils.get_project_root(), "tests/test_resources")
+TEST_MESHES = [os.path.join(TEST_RESOURCES, glb) for glb in os.listdir(TEST_RESOURCES) if glb.endswith(".glb")]
+
+
+@pytest.mark.parametrize("mesh_fpath", TEST_MESHES)
+def test_repair_mesh(mesh_fpath: str):
+    # Load up the mesh
+    loaded = load_mesh(mesh_fpath)
+    in_faces = loaded.faces.copy()
+    # Repair the mesh, in-place
+    new_mesh = validate_mesh(loaded)
+    out_faces = new_mesh.faces
+    # Should have a different number of faces after repair
+    assert len(in_faces) != len(out_faces)
+
+
+@pytest.mark.parametrize("mesh_fpath", TEST_MESHES)
+def test_load_mesh(mesh_fpath: str):
+    loaded = load_mesh(mesh_fpath)
+    assert isinstance(loaded, Trimesh)
+    assert loaded.metadata["fpath"] == mesh_fpath
+
+
+@pytest.mark.parametrize("mesh_fpath,expected", [("iamnotafile", FileNotFoundError), (1234, TypeError)])
+def test_load_broken_mesh(mesh_fpath: str, expected):
+    with pytest.raises(expected):
+        load_mesh(mesh_fpath)
+
+
+@pytest.fixture(scope="module")
+def oyens_space() -> Space:
+    """Returns a Space object with the Oyens mesh (Gibson) and a single microphone"""
+    oyens = os.path.join(utils.get_project_root(), "tests/test_resources/Oyens.glb")
+    space = Space(
+        oyens,
+        mic_positions=[-0.5, -0.5, 0.5],
+        min_distance_from_source=0.2,
+        min_distance_from_mic=0.1,
+        min_distance_from_surface=0.2
+    )
+    return space
+
+
+@pytest.mark.parametrize(
+    "microphone_positions",
+    [
+        np.array([[-0.5, -0.5, 0.5,]]),    # 1 mic setup
+        np.array([[-0.1, -0.1, 0.6], [0.5, 0.5, 0.5]])     # 2 mic setup
+    ]
+)
+def test_place_microphones(microphone_positions):
+    space = Space(TEST_MESHES[0], microphone_positions)
+    # Should have the same number of mic positions we've passed in
+    assert isinstance(space.mic_positions, np.ndarray)
+    # All mic positions should be in XYZ format
+    assert space.mic_positions.shape == (len(microphone_positions), 3)
```

Comment:
Just for full transparency: this tests that each microphone is placed with 3D coordinates, but not necessarily in XYZ order.